drop-in/src/main/java/com/adyen/checkout/dropin/ui/action/ActionComponentDialogFragment.kt

```diff
@@ -151,11 +161,17 @@ internal class ActionComponentDialogFragment : DropInBottomSheetDialogFragment()
     }
 
     private fun handleError(componentError: ComponentError) {
-        when (componentError.exception) {
+        when (val exception = componentError.exception) {
             is CancellationException -> {
                 Logger.d(TAG, "Flow was cancelled by user")
                 onBackPressed()
             }
+            is PermissionException -> {
+                val requiredPermission = exception.requiredPermission
+                Logger.d(TAG, "$requiredPermission permission not granted")
+                exception.message?.let { requireContext().toast(it) }
```

Comment:
The toast message here should probably be a dialog, with `requestPermissionLauncher.launch(requiredPermission)` called in `OnDismissListener`. And the message should not be coming from `exception.message` but from the strings.

Reply:
I talked with Arjen about that and we will just leave the storage native dialog and will remove the toast.